src/lib.rs

```diff
@@ -198,7 +198,7 @@ fn counter_high(counter: u64) -> u32 {
 /// [`from_hex`]: #method.from_hex
 /// [`Display`]: https://doc.rust-lang.org/std/fmt/trait.Display.html
 /// [`FromStr`]: https://doc.rust-lang.org/std/str/trait.FromStr.html
-#[derive(Clone, Copy, Hash)]
+#[derive(Clone, Copy)]
```

Comment:
I understand the general intent of this lint, but in this case I think `derive(Hash)` is fine. Any two BLAKE3 hashes that compare equal will also give the same HashMap hash. (The current `PartialEq` implementation has the same behavior that `derive(PartialEq)` would, with the important difference being that it's done in constant time.) More importantly, removing `derive(Hash)` here without an explicit impl to replace it would be a breaking change.